nncf/quantization/algorithms/min_max/torch_backend.py

```diff
@@ -155,32 +156,50 @@ def get_statistic_collector(
         quantizer_config: QuantizerConfig,
         inplace: bool,
         num_samples: int = None,
-    ) -> Union[PTMinMaxStatisticCollector, PTMeanMinMaxStatisticCollector]:
-        if (
-            range_estimator_params.min.statistics_type == StatisticsType.MIN
-            and range_estimator_params.min.aggregator_type == AggregatorType.MIN
-            and range_estimator_params.max.statistics_type == StatisticsType.MAX
-            and range_estimator_params.max.aggregator_type == AggregatorType.MAX
-        ):
-            collector_name = "min_max"
-
-        elif (
-            range_estimator_params.min.statistics_type == StatisticsType.MIN
-            and range_estimator_params.min.aggregator_type == AggregatorType.MEAN
-            and range_estimator_params.max.statistics_type == StatisticsType.MAX
-            and range_estimator_params.max.aggregator_type == AggregatorType.MEAN
+    ) -> TensorCollector:
+        collector_params = PTMinMaxAlgoBackend._default_collector_params(nncf_graph, target_point, quantizer_config)
+        collector_kwargs = collector_params.convert_statistic_params(per_sample_stats=False)
+
+        collector = TensorCollector(PTMinMaxTensorStatistic)
+        for params, container_key in zip(
+            [range_estimator_params.min, range_estimator_params.max],
+            [PTMinMaxTensorStatistic.MIN_STAT, PTMinMaxTensorStatistic.MAX_STAT],
         ):
-            collector_name = "mean_min_max"
-
-        else:
-            raise RuntimeError(
-                "The following range estimator parameters are not supported by PyTorch backend by now: "
-                f"{str(range_estimator_params)}"
-            )
-
-        return PTMinMaxAlgoBackend._statistic_collector_builder(
-            collector_name, nncf_graph, target_point, quantizer_config, num_samples
-        )
+            if not params.statistics_type in PT_REDUCERS_MAP:
+                raise RuntimeError(
+                    f"Statistic type: {params.statistics_type} is not supported for Torch PTQ backend yet."
+                )
+
+            if not params.aggregator_type in AGGREGATORS_MAP:
+                raise RuntimeError(
+                    f"Aggregator type: {params.aggregator_type} is not supported for Torch PTQ backend yet."
+                )
+
+            kwargs = {
+                "reduction_axes": collector_kwargs["reducers_axes"],
+            }
```

Comment:
Do you suggest to make a different call for each set of attributes? So, for quantiles it needs quantile, for others is not. What are disadvantages of this method?

Reply:
I have answered this before already. You are forfeiting both readability and IDE support just to save several lines of code.

The problem of instantiating a list of inhomogenous objects with different sets of parameters in one place is a classical problem:
https://softwareengineering.stackexchange.com/questions/389503/should-i-use-the-factory-pattern-when-instantiating-objects-with-very-different

What is being done in this function is basically a factory. Since the constructors for the collectors do not follow a common signature, you will have to do a switch for all possible collectors, writing out the instantiation and the parameter list explicitly, and there is no way around it. If you find yourself doing this too often, then in order not to duplicate the code you should be extracting the functionality into an actual factory class, parametrized on `RangeEstimatorParameters`, for instance. Basically, doing anything like `PT_REDUCERS_MAP[statistic_type](**kwargs)` because you can never tell if the `kwargs` are correct for a given `PT_REDUCERS_MAP[statistic_type]` outside of running the program in all possible situations. With an explicit switch, you can be sure that the parameter names, parameter count and parameter types are correct even before running tests.

Reply:
Got your point. Done